languages/haskell/docs/implementing-a-concept-exercise.md

```diff
@@ -1,5 +1,53 @@
-# How to implement an Haskell concept exercise
+# How to implement a Haskell concept exercise
 
-TODO: describe how to implement a concept exercise for the Haskell track. For inspiration, check out the [C# version of this file][csharp-implementing].
+This document describes the steps required to implement a concept exercise for the Haskell track. As this document is generic, the following placeholders are used:
 
-[csharp-implementing]: ../../csharp/docs/implementing-a-concept-exercise.md
+- `<SLUG>`: the name of the exercise in kebab-case (e.g. `anonymous-methods`).
+- `<NAME>`: the name of the exercise in PascalCase (e.g. `AnonymousMethods`).
+
+Before implementing the exercise, please make sure you have a good understanding of what the exercise should be teaching (and what not). This information can be found in the exercise's GitHub issue. Concept exercises should be modelled after the CIS 194 Haskell course at Penn University, either the [Spring 2013][spring-2013] or [Fall 2016][fall-2016] version.
+
+To implement a concept exercise,  the following files need to be created:
+
+<pre>
+languages
+└── Haskell
+    └── exercises
+        └── concept
+            └── <SLUG>
+                ├── .docs
+                |   ├── instructions.md
+                |   ├── introduction.md
+                |   ├── hints.md
+                |   └── after.md (optional)
+                ├── .meta
+                |   ├── config.json
+                |   ├── <SLUG>.cabal
+                |   ├── stack.yaml
+                |   └── Example.hs
+                ├── src
+                |   └── <NAME>.hs
+                └── test
+                    └── Main.hs
+</pre>
+
+## Step 1: add track-specific files
```

Comment:
Isn't `<SLUG>.cabal` also a language-specific file?

Reply:
It's automatically generated by package.yaml apparently:

```cabal
-- This file has been generated from package.yaml by hpack version 0.31.2.
--
-- see: https://github.com/sol/hpack
--
-- hash: 2023dceec3fe5a747e981d5a0d4cfc6a111057404efa2ab8be9fe514b6c3a4b7
```
^ Current Haskell Leap exercise cabal file (leap.cabal)

Reply:
As such I've removed it from the directory tree.

Reply:
OK, this resolves one of my questions. I'll probably create a thread that addresses when this was last decided and reiterate our reasons. `package-yaml`-only projects are a sense more opinioned (people *have* to use Stack) than the alternatives. With Cabal's recent [common stanzas](https://vrom911.github.io/blog/common-stanzas) some of the reasons for using `package.yaml` have been reclaimed, but not all.